File: ExpressionTreeTransform.Tests/Runners.cs
using System;
using System.Linq.Expressions;
using Xunit;
using static ExpressionTreeTransform.Util.Globals;

namespace ExpressionTreeTransform.Tests {
    public static class Runners {
        public static void BuildAssert(Expression<Action> expr, string csharp, string vb)  => BuildAssert(expr as Expression, csharp, vb);
        public static void BuildAssert<T>(Expression<Action<T>> expr, string csharp, string vb) => BuildAssert(expr as Expression, csharp, vb);
        public static void BuildAssert<T1, T2>(Expression<Action<T1, T2>> expr, string csharp, string vb) => BuildAssert(expr as Expression, csharp, vb);

        public static void BuildAssert<T>(Expression<Func<T>> expr, string csharp, string vb) => BuildAssert(expr as Expression, csharp, vb);
        public static void BuildAssert<T1, T2>(Expression<Func<T1, T2>> expr, string csharp, string vb) => BuildAssert(expr as Expression, csharp, vb);
        public static void BuildAssert<T1, T2, T3>(Expression<Func<T1, T2, T3>> expr, string csharp, string vb) => BuildAssert(expr as Expression, csharp, vb);

        public static void BuildAssert(Expression expr, string csharp, string vb) {
            var testCSharpCode = expr.ToCode(CSharp);
            var testVBCode = expr.ToCode(VisualBasic);
            Assert.True(testCSharpCode == csharp);
            Assert.True(testVBCode == vb);
        }
    }
}

File: ExpressionTreeTransform.Tests.VB/Runners.vb
Imports System.Linq.Expressions

<Assembly: AssemblyTrait("Source", VBCompiler)>

Public Module Runners
    Sub BuildAssert(Of T)(expr As Expression(Of Func(Of T)), CSharpResult As String, VBResult As String)
        BuildAssert(CType(expr, Expression), CSharpResult, VBResult)
    End Sub

    Sub BuildAssert(expr As Expression, CSharpResult As String, VBResult As String)
        Dim testCSharpCode = expr.ToCode(CSharp)
        Dim testVBCode = expr.ToCode(VisualBasic)
        Assert.True(testCSharpCode = CSharpResult)
        Assert.True(testVBCode = VBResult)
    End Sub
End Module
